docs/operator-manual/pprof.md

```diff
@@ -0,0 +1,38 @@
+# Profiling with pprof
+
+pprof is a powerful tool for profiling Go applications. It is part of the Go standard library and provides rich insights into the performance characteristics of Go programs. Profiling with pprof involves gathering runtime statistics about the application's execution, such as CPU usage, memory allocation, and contention.
+
+## Installation
+
+pprof is included in the Go standard library and integrated in all ArgoCD components, so no separate installation is required. To enable the profiling endpoints, you just need to set the environment variable `ARGO_PPROF` with the port that you want to use for exposing the endpoints. e.g: `ARGO_PPROF=8888` will start the profiling endpoints using the port `8888`.
+
+## Basic Usage
+pprof operates in two modes: interactive and non-interactive. In the non-interactive mode, it generates profiling data that can be analyzed later. In the interactive mode, it launches a web server to visualize the profiling data in real-time.
+
+> Note: Replace http://localhost:6060 with the appropriate URL of your Go application.
+
+### Generating CPU Profiles
+To generate a CPU profile for a Go application, you can use the following command:
```

Comment:
```suggestion
### Generate CPU Profile

Generate a CPU profile with the following command:
```

- use active voice, per k8s style guide
- remove "for a Go application"; not needed
- both of these make it simpler and more direct, per k8s style guide

- also newline after heading, per `markdownlint`